Ginger/GingerCoreNET/ActionsLib/ActionConversionUtils.cs

```diff
@@ -0,0 +1,192 @@
+﻿using amdocs.ginger.GingerCoreNET;
+using Amdocs.Ginger.Common;
+using Amdocs.Ginger.Common.UIElement;
+using Amdocs.Ginger.Repository;
+using GingerCore.Actions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace Amdocs.Ginger.CoreNET
+{
+    /// <summary>
+    /// This class is used to add methods for action conversion helpers
+    /// </summary>
+    public class ActionConversionUtils
+    {
+        /// <summary>
+        /// private constructor
+        /// </summary>
+        private ActionConversionUtils()
+        {
+        }
+
+        private static ActionConversionUtils mInstance;
+        public static ActionConversionUtils Instance
+        {
+            get
+            {
+                if(mInstance == null)
+                {
+                    mInstance = new ActionConversionUtils();
+                }
+                return mInstance;
+            }
+        }
+
+        /// <summary>
+        /// This method is used to find the relative element from POM for the existing action
+        /// </summary>
+        /// <param name="newActUIElement"></param>
+        /// <param name="pomModelObject"></param>
+        /// <returns></returns>
+        public Act GetMappedElementFromPOMForAction(Act newActUIElement, string pomModelObject)
+        {
+            ApplicationPOMModel selectedPOM = new ApplicationPOMModel();
+
+            ObservableList<ApplicationPOMModel> pomLst = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ApplicationPOMModel>();
+            selectedPOM = pomLst.Where(x => x.Guid.ToString() == pomModelObject).SingleOrDefault();
+
+            bool isPOM = true;
+            ElementInfo elementInfo = null;
+            IEnumerable<ElementInfo> lst = selectedPOM.MappedUIElements.Where(x => x.ElementTypeEnum != eElementType.Div);
+
+            string locateValue = Convert.ToString(newActUIElement.GetType().GetProperty("LocateValue").GetValue(newActUIElement, null));
+            string elementLocateBy = Convert.ToString(newActUIElement.GetType().GetProperty("ElementLocateBy").GetValue(newActUIElement, null));
```

Comment:
"elementLocateBy" should be enum and not string

Reply:
Done